src/Middleware/Rewrite/src/RewriteOptions.cs

```diff
@@ -20,6 +22,6 @@ public class RewriteOptions
         /// <summary>
         /// Gets and sets the File Provider for file and directory checks. Defaults to <see cref="IHostingEnvironment.WebRootFileProvider"/>
         /// </summary>
-        public IFileProvider StaticFileProvider { get; set; }
+        public IFileProvider StaticFileProvider { get; set; } = default!;
```

Comment:
```suggestion
        public IFileProvider? StaticFileProvider { get; set; }
```

Reply:
With the way options setup works, this type is initialized to a non-null value before users get to use it. While it's true it's null if you new up the options type, we've generally documented the most common behavior.

I'll leave it up to @Tratcher to make a call though

Reply:
If users never see null then there's no point in claiming it can be null.